fastlane/swift/ConfigItem.swift

```diff
@@ -0,0 +1,129 @@
+// ConfigItem.swift
+// Copyright (c) 2021 FastlaneTools
+
+//
+//  ** NOTE **
+//  This file is provided by fastlane and WILL be overwritten in future updates
+//  If you want to add extra functionality to this project, create a new file in a
+//  new group so that it won't be marked for upgrade
+//
+
+import Foundation
+
+public enum ConfigItem<T> {
+    case fastlaneDefault(T)
+    case userDefined(T)
+    case `nil`
```

Comment:
I don't fully understand this yet but could it be possible that we use only two cases; `fastlaneDefault` and `userDefined`?

Reply:
`case nil` serves the purpose of filling the case where the user defines as `nil` on-hand but that shouldn't be sent to the Ruby side

Reply:
And as the extensions are being set over concrete non-optional values, we have to set at some point the `ExpressibleByNilLiteral` to be set to that extra `nil` case.